codex/contracts/provider.nim

```diff
@@ -0,0 +1,131 @@
+import pkg/ethers/provider
+import pkg/chronos
+import pkg/questionable
+
+import ../logutils
+
+from ../errors import CodexError
+from ../clock import SecondsSince1970
+
+logScope:
+  topics = "marketplace onchain provider"
+
+type CodexProviderError* = object of CodexError
+
+proc raiseCodexProviderError(message: string) {.raises: [CodexProviderError].} =
+  raise newException(CodexProviderError, message)
```

Comment:
I like your suggestion. I was struggling a bit how to look at it. I will then raise the original `ProviderError` from `nim-ethers`. This will also simplify `raises` annotations.